src/EnsembleKalmanProcess.jl

```diff
@@ -336,4 +368,317 @@ function update_ensemble!(ekp::EnsembleKalmanProcess{FT, IT, Sampler{FT}}, g_in:
+    
+    c_weights = process.c_weights
+    
+    
+    chol_xx_cov = cholesky(Hermitian(x_cov)).L
+    
+    x = zeros(FT, N_x, 2*N_x+1)
+    x[:, 1] = x_mean
+    for i = 1: N_x
+        x[:, i+1] = x_mean + c_weights[i]*chol_xx_cov[:,i]
+        x[:, i+1+N_x] = x_mean - c_weights[i]*chol_xx_cov[:,i]
+    end
+    
+    return x
+end
+
+
+"""
+construct_mean x_mean from ensemble x
+"""
+function construct_mean(uki::EnsembleKalmanProcess{FT, IT,Unscented}, x::Array{FT,2}) where {FT<:AbstractFloat, IT<:Int}
+    N_x, N_ens = size(x)
+    
+    @assert(uki.N_ens == N_ens)
+    
+    x_mean = zeros(FT, N_x)
+    
+    mean_weights = uki.process.mean_weights
+    
+    
+    for i = 1: N_ens
+        x_mean += mean_weights[i]*x[:, i]
+    end
+    
+    return x_mean
+end
+
+"""
+construct_cov xx_cov from ensemble x and mean x_mean
+"""
+function construct_cov(uki::EnsembleKalmanProcess{FT, IT,Unscented}, x::Array{FT,2}, x_mean::Array{FT}) where {FT<:AbstractFloat, IT<:Int}
+    N_ens, N_x = uki.N_ens, size(x_mean,1)
+    
+    cov_weights = uki.process.cov_weights
+    
+    xx_cov = zeros(FT, N_x, N_x)
+    
+    for i = 1: N_ens
+        xx_cov .+= cov_weights[i]*(x[:,i] - x_mean)*(x[:,i] - x_mean)'
+    end
+    
+    return xx_cov
+end
+
+"""
+construct_cov xy_cov from ensemble x and mean x_mean, ensemble obs_mean and mean y_mean
+"""
+function construct_cov(uki::EnsembleKalmanProcess{FT, IT, Unscented}, x::Array{FT,2}, x_mean::Array{FT}, obs_mean::Array{FT,2}, y_mean::Array{FT}) where {FT<:AbstractFloat, IT<:Int, P<:Process}
+    N_ens, N_x, N_y = uki.N_ens, size(x_mean,1), size(y_mean,1)
+    
+    cov_weights = uki.process.cov_weights
+    
+    xy_cov = zeros(FT, N_x, N_y)
+    
+    for i = 1: N_ens
+        xy_cov .+= cov_weights[i]*(x[:, i] - x_mean)*(obs_mean[:, i] - y_mean)'
+    end
+    
+    return xy_cov
+end
+
+"""
+uki prediction step : generate sigma points
+"""
+function update_ensemble_prediction!(process::Unscented) where {FT<:AbstractFloat, IT<:Int}
+    
+    process.iter += 1
+    # update evolution covariance matrix
+    if process.update_freq > 0 && process.iter%process.update_freq == 0
+        process.Σ_ω = (2 - process.α_reg^2)process.uu_cov[end]
+    end
+    
+    u_mean  = process.u_mean[end]
+    uu_cov = process.uu_cov[end]
+    
+    α_reg = process.α_reg
+    r = process.r
+    Σ_ω = process.Σ_ω
+    
+    N_u = length(process.u_mean[1])
+    ############# Prediction step:
+    
+    u_p_mean  = α_reg*u_mean + (1-α_reg)*r
+    uu_p_cov = α_reg^2*uu_cov + Σ_ω
+    
+    ############ Generate sigma points
+    u_p = construct_sigma_ensemble(process, u_p_mean, uu_p_cov)
+    return u_p
+end
+
+
+"""
+uki analysis step 
+g is the predicted observations  Ny  by N_ens matrix
+"""
+function update_ensemble_analysis!(uki::EnsembleKalmanProcess{FT, IT,Unscented}, u_p::Array{FT, 2}, g::Array{FT, 2}) where {FT<:AbstractFloat, IT<:Int}
+    
+    obs_mean = uki.obs_mean
+    Σ_ν = uki.process.Σ_ν
```

Comment:
If we store `Σ_ν` as the constant `2` in the `Unscented` object instead of `2*obs_cov_noise`, then this can simply be replaced by 
`Σ_ν = uki.process.Σ_ν * uki.obs_noise_cov`